src/ray/object_manager/object_manager.cc

```diff
@@ -398,15 +405,19 @@ ray::Status ObjectManager::SendObjectHeaders(const ObjectID &object_id,
 
   // Fail on status not okay. The object is local, and there is
   // no other anticipated error here.
-  RAY_CHECK_OK(chunk_status.second);
+  ray::Status status = chunk_status.second;
+  if (!chunk_status.second.ok()) {
+    RAY_LOG(WARNING) << "Attempting to push object " << object_id
+                     << " which is not local. It may have been evicted.";
+    RAY_RETURN_NOT_OK(status);
```

Comment:
This makes sense, but it seems we may need to introduce a condition into `ObjectBufferPool::GetChunk` to ensure that if `GetChunk` is invoked on object `x`, and object `x` is being created, then we return a bad status.

If the check in `plasma/client.cc` (which ensures no object is read until sealed) is no longer fatal (e.g. it returns a bad status), I think `ObjectBufferPool::GetChunk` still needs to be updated to do more than just `ARROW_CHECK_OK` on what is returned by `store_client_.Get`.

Reply:
So, currently when that happens, the client just says that the object is not present, so it should go through this code path.

https://github.com/ray-project/ray/blob/8d8b6e5bfab1993718580205612e5397215a1bd6/src/ray/object_manager/object_buffer_pool.cc#L46-L51